### PR TITLE
Fix repo validation 2

### PR DESCRIPTION
**Pull request by eloots**
Second part of student repo validation adaptation:

- Add `ICselectFirstExercise` command to studentiiied repository
- Utilize `ICselectFirstExercise` in `validateStudentRepo.sh` script

Verified this on FTTAS, FTTS, AS, FTTASS and AAS with first exercise having a non-zero value

### CODE REVIEW

File: sbtStudentCommands/IC.scala.template [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import sbt._
+import Keys._
+import sbt.complete.DefaultParsers._
+
+import scala.Console
+import StudentKeys._
+import Zip.withZipFile
+import sbtstudent.Pssr.{getCueFolder, getRootFolder}
+
+object IC {
```
Comment (WadeWaldron) [inline sbtStudentCommands/IC.scala.template]: What is IC?

File: sbtStudentCommands/IC.scala.template [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def ICselectFirstExercise: Command = Command.command("ICselectFirstExercise") { state =>
+
+    val firstExercise = state.get(mapNext).get.keys.toVector.sorted.head
+    withZipFile(state, firstExercise) { () =>
+      val currentExerciseMain = new sbt.File(getRootFolder(state), "exercises")
+      val cueFolder = getCueFolder(state)
+      val solFolder = new sbt.File(cueFolder, firstExercise)
+      IO.delete(currentExerciseMain)
+      IO.copyDirectory(solFolder, currentExerciseMain)
```
Comment (WadeWaldron) [inline sbtStudentCommands/IC.scala.template]: Why do we pull the solution? Shouldn't the command simply take us to the first exercise, then give us the option to pull the solution?

The validate script, where this is used, is going to pull the solution anyway.

File: validateStudentRepo.sh [hunk 3]
```diff
@@ -37,16 +39,15 @@ function validateAllExercises {
     cd $studentRepo
 
     echo $SEPARATOR
-    sbt "gotoExerciseNr 0" 2>&1 | grep "Moved to"
```
Comment (WadeWaldron) [inline validateStudentRepo.sh]: Are there any courses where gotoExerciseNr 0 doesn't work exactly the same as ICselectFirstExercise?

**Comment by eloots** [comment]
Internal Command

Groeten, Eric

File: validateStudentRepo.sh (hunk 3 above)
Comment (eloots) [inline validateStudentRepo.sh]: ?

Comment (eloots) [inline validateStudentRepo.sh]: ICselectFirstExercise semantics is fundamentally different from gotoExercise

Comment (WadeWaldron) [inline validateStudentRepo.sh]: How so? Perhaps I am not understanding what this does then.

File: sbtStudentCommands/IC.scala.template (hunk 2 above)
Comment (eloots) [inline sbtStudentCommands/IC.scala.template]: You do have a point about the fact that the solution shouldn't be pulled (as validateStudentRepo.sh does this just before running `test`

File: validateStudentRepo.sh (hunk 3 above)
Comment (eloots) [inline validateStudentRepo.sh]: Just made a number of modifications.
`ICselectFirstExercise` has been modified as it didn't update the exercises bookmark and it unnecessarily pulled the solution for the first exercise. Instead now:

- It now just sets the exercises bookmark to the first exercise in the course.
- It is renamed to reflect what it actually does: `ICSetBookmarkToFirstExercise`

**Comment by eloots** [comment]
@WadeWaldron @sebastianharko Documented the `gotoFirstExercise` command so that it can be utilised by anyone who might find it useful. Please review.

File: sbtStudentCommands/Navigation.scala.template [hunk 4]
```diff
@@ -104,6 +104,25 @@ object Navigation {
 
   private lazy val ExerciseNr = OptSpace ~> IntBasic.?
 
+  def gotoFirstExercise: Command = Command.command("gotoFirstExercise") { state =>
+    val firstExercise = state.get(mapNext).get.keys.toVector.sorted.head
+    withZipFile(state, firstExercise) { () =>
+      val src = new sbt.File(new sbt.File(Project.extract(state).structure.root), "exercises")
```
Comment (WadeWaldron) [inline sbtStudentCommands/Navigation.scala.template]: Guessing this is basically a copy/paste from gotoExerciseNr. Goto First Exercises should probably just figure out the exercise number and then call gotoExerciseNr under the hood to eliminate duplication.

**Pull request merged**